src/ctapipe/monitoring/aggregator.py

```diff
@@ -1,159 +1,390 @@
+        """
+        Compute statistics and add columns to results dictionary.
+
+        This method should compute aggregator-specific statistics from the data
+        and append values to appropriate keys in results_dict.
+
+        The base implementation already adds: time_start, time_end, event_id_start, event_id_end.
+        Subclasses should add their specific statistics (e.g., mean, median, std, n_events, etc.).
+
+        Parameters
+        ----------
+        data : array-like
+            Data for this chunk (already extracted from table)
+        masked_elements_of_sample : ndarray, optional
+            Boolean mask of shape (\*data_dimensions) for elements to exclude
+        results_dict : dict
+            Dictionary to which statistic columns should be added.
+            Use results_dict['column_name'].append(value) for each statistic.
+
+        Examples
+        --------
+        # Add mean, median, std columns:
+        results_dict['mean'].append(np.mean(data))
+        results_dict['median'].append(np.median(data))
+        results_dict['std'].append(np.std(data))
+        """
+        pass
+
+    @abstractmethod
+    def _set_result_units(self, table, unit):
+        """
+        Set units for result columns that should inherit from the input data.
+
+        This method is called after the result table is created to assign units
+        to columns that were computed from the input data and should have the same units.
+
+        Parameters
+        ----------
+        table : astropy.table.Table
+            The result table with computed statistics
+        unit : astropy.units.Unit
+            The unit from the input data column
+
+        Examples
+        --------
+        # Set units for specific columns:
+        table['mean'].unit = unit
+        table['median'].unit = unit
+        table['std'].unit = unit
+        """
+        pass
+
+    def _get_chunks(self, table, chunk_shift, effective_chunk_size):
+        """
+        Generate chunks from table based on chunking strategy.
+
+        Parameters
+        ----------
+        table : astropy.table.Table
+            Input table to chunk
+        chunk_shift : int or None
+            Number of events/seconds to shift between chunks
+        effective_chunk_size : int
+            Size of each chunk in events or seconds
+
+        Yields
+        ------
+        astropy.table.Table
+            Chunks of the input table
+        """
+        # If using entire table as one chunk, just yield the whole table
+        if self.chunk_size is None:
+            yield table
+            return
+
+        if self.chunking_mode == "events":
+            yield from self._get_event_chunks(table, chunk_shift, effective_chunk_size)
+        elif self.chunking_mode == "time":
+            yield from self._get_time_chunks(table, chunk_shift, effective_chunk_size)
+
+    def _check_table_length(self, table, effective_chunk_size):
```

Comment:
I'd consider moving this check into `_get_event_chunks()` and `_get_time_chunks()` and remove this extra function.